src/Package/Impl/Repl/Commands/WorkingDirectoryCommand.cs

```diff
@@ -163,24 +164,11 @@ public sealed class WorkingDirectoryCommand : Command, IDisposable {
             return PathHelper.MakeRooted(PathHelper.EnsureTrailingSlash(UserDirectory), friendlyName.Substring(2));
         }
 
-        internal async Task<string> GetRWorkingDirectoryAsync() {
-            await TaskUtilities.SwitchToBackgroundThread();
-            try {
-                using (var evaluation = await _session.BeginEvaluationAsync(false)) {
-                    return await evaluation.GetWorkingDirectory();
-                }
-            } catch (TaskCanceledException) { }
-            return null;
-        }
-
         private async Task UpdateRUserDirectoryAsync() {
             await TaskUtilities.SwitchToBackgroundThread();
             try {
-                using (var evaluation = await _session.BeginEvaluationAsync(false)) {
-                    UserDirectory = await evaluation.GetRUserDirectory();
-                }
-            } catch (TaskCanceledException) {
-            }
+                UserDirectory = await SessionUtilities.GetRUserDirectoryAsync(_interactiveWorkflow);
+            } catch (TaskCanceledException) { }
```

Comment:
Actually, now that I look at it, `GetRUserDirectoryAsync` already swallows exceptions (should it?).